### PR TITLE
1.3.16-pre: 1.3.15 Fixes

### PR DESCRIPTION
**Pull request by daxgames**
### Fixes

* #2340 `GIT_INSTALL_ROOT` gets set incorrectly to `git_root\mingw64\bin` path
* [#2299 ](https://github.com/cmderdev/cmder/issues/2299#issuecomment-644379573) `fast_init` checked for too soon, which results in a bunch of debug errors in certain fringe conditions.\

## Adds

* Made [`%cmder_root%\vendor\bin\vscode_init.bat`](https://github.com/cmderdev/cmder/blob/f1c1354c007c7018cd31493e0a18dc602b694bb5/vendor/bin/vscode_init.cmd) smarter so it can configure Cmder integrated terminal whether it is launched inside or outside cmder.

### CODE REVIEW

File: vendor/init.bat [hunk 1]
```diff
@@ -200,14 +201,14 @@ if defined GIT_INSTALL_ROOT (
...
+    if not defined GIT_INSTALL_ROOT (
+        :: get the absolute path to the user provided git binary
+        %lib_git% is_git_shim "%%~dpF"
+        %lib_git% get_user_git_version
+        %lib_git% compare_git_versions
 
-if defined GIT_INSTALL_ROOT (
-    goto :FOUND_GIT
+        goto :FOUND_GIT
+    )
```
Comment (jack1142) [inline vendor/init.bat:211]: This makes the script always go to `FOUND_GIT` which does not seem intended, e.g. `%lib_git% compare_git_versions` does not define `GIT_INSTALL_ROOT` when the checked git is older than the vendored git.

So this should probably be checked at the end of each iteration instead:
```suggestion
    :: get the absolute path to the user provided git binary
    %lib_git% is_git_shim "%%~dpF"
    %lib_git% get_user_git_version
    %lib_git% compare_git_versions

    if defined GIT_INSTALL_ROOT (
        goto :FOUND_GIT
    )
```

**Comment by daxgames** [comment]
@jack1142 you are correct.  Fixed

**Comment by daxgames** [comment]
New build https://ci.appveyor.com/project/MartiUK/cmder/builds/33845600/artifacts

**Comment by daxgames** [comment]
I need to look at one other thing and when I do I will merge this in.

**Pull request merged**